src/main/java/de/tum/cit/aet/artemis/programming/web/ExerciseSharingResource.java

```diff
@@ -0,0 +1,212 @@
+
+    private static final Logger log = LoggerFactory.getLogger(ExerciseSharingResource.class);
+
+    private final ExerciseSharingService exerciseSharingService;
+
+    private final SharingConnectorService sharingConnectorService;
+
+    private final ProgrammingExerciseImportFromSharingService programmingExerciseImportFromSharingService;
+
+    /**
+     * constructor for spring
+     *
+     * @param exerciseSharingService                      the sharing service
+     * @param sharingConnectorService                     the sharing connector service
+     * @param programmingExerciseImportFromSharingService programming exercise import from sharing service
+     */
+    public ExerciseSharingResource(ExerciseSharingService exerciseSharingService, SharingConnectorService sharingConnectorService,
+            ProgrammingExerciseImportFromSharingService programmingExerciseImportFromSharingService) {
+        this.exerciseSharingService = exerciseSharingService;
+        this.programmingExerciseImportFromSharingService = programmingExerciseImportFromSharingService;
+        this.sharingConnectorService = sharingConnectorService;
+    }
+
+    /**
+     * GET .../sharing-import/basket
+     *
+     * @param basketToken the token of the shopping basket
+     * @param returnURL   the URL to return to after the basket is loaded
+     * @param apiBaseURL  the base URL of the API
+     * @param checksum    the checksum to validate the request
+     * @return the ResponseEntity with status 200 (OK) and with body the Shopping Basket, or with status 404 (Not Found)
+     */
+    @GetMapping("import/basket")
+    public ResponseEntity<ShoppingBasket> loadShoppingBasket(@RequestParam String basketToken, @RequestParam String returnURL, @RequestParam String apiBaseURL,
+            @RequestParam String checksum) {
+        if (SecretChecksumCalculator.checkChecksum(Map.of("returnURL", returnURL, "apiBaseURL", apiBaseURL), sharingConnectorService.getSharingApiKeyOrNull(), checksum)) {
+            Optional<ShoppingBasket> sharingInfoDTO = exerciseSharingService.getBasketInfo(basketToken, apiBaseURL);
+            return ResponseUtil.wrapOrNotFound(sharingInfoDTO);
+        }
+        else {
+            log.warn("Checksum validation failed for basketToken={}", basketToken);
+            return ResponseEntity.badRequest().build();
+        }
+    }
+
+    /**
+     * GET .../sharing/setup-import
+     * sets up the imported exercise as a ProgrammingExercise.
+     *
+     * @param sharingSetupInfo the sharing setup information containing the details of the exercise to be imported
+     * @throws GitAPIException if there is an error with the git operations
+     * @return the ResponseEntity with status 200 (OK) and with body the programming exercise, or with status 404 (Not Found)
+     */
+    @PostMapping("setup-import")
+    public ResponseEntity<ProgrammingExercise> setUpFromSharingImport(@RequestBody SharingSetupInfo sharingSetupInfo)
```

Comment:
❗ Missing authorization for the API endpoint. There is no check that only editors/instructors of the target course can trigger the import.

Reply:
❗ Similar issue for other API endpoints in the class, too.